lib/pos_network_message_interface.go

```diff
@@ -14,6 +15,39 @@ import (
 // place all translations between types defined in lib and consensus packages.  //
 //////////////////////////////////////////////////////////////////////////////////
 
+// MsgDeSoHeader struct <-> consensus.Block interface translation
+
+func (msg *MsgDeSoHeader) GetBlockHash() consensus.BlockHash {
+	hash, err := msg.Hash()
+	if err != nil {
+		glog.Errorf("MsgDeSoHeader.GetBlockHash: Problem hashing header: %v", err)
+		// If we can't generate the block hash, return nil and have the caller handle it.
+		return nil
```

Comment:
I think this is okay but we need to check all the callers to make sure we're not introducing a panic anywhere.

Reply:
The consensus module validates incoming blocks (and their block hashes) on input. This is safe